packages/shared/src/hdf5-models.ts

```diff
@@ -207,21 +207,19 @@ export interface UnknownType {
 /* ----------------- */
 /* ----- VALUE ----- */
 
-export type Primitive<T extends DType> = T extends NumericType | EnumType
-  ? number
+export type Primitive<T extends DType = DType> = T extends NumericLikeType
+  ? number | (T extends BooleanType ? boolean : never) // let providers choose how to return booleans
```

Comment:
Just a small refactoring to make use of `NumericLikeType` and avoid duplicating `number`.